src/test/java/easytests/core/services/TopicsServiceTest.java

```diff
@@ -3,100 +3,261 @@
+
+        verify(this.topicsMapper, times(1)).insert(topicEntityCaptor.capture());
+        this.topicsSupport.assertEquals(this.topicsSupport.getEntityAdditionalMock(0), topicEntityCaptor.getValue());
+    }
+
+    @Test
+    public void testSaveUpdatesEntityIdOnCreation() throws Exception {
+        final TopicModelInterface topicAdditionalModel = this.topicsSupport.getModelAdditionalMock(0);
+        final Integer updatedId = 10;
+        doAnswer(invocation -> {
+            final TopicEntity topicEntity = invocation.getArgument(0);
+            topicEntity.setId(updatedId);
+            return null;
+        }).when(this.topicsMapper).insert(any(TopicEntity.class));
+
+        this.topicsService.save(topicAdditionalModel);
+
+        verify(topicAdditionalModel, times(1)).setId(updatedId);
+    }
+
+    @Test
+    public void testSaveUpdatesEntity() throws Exception {
+        final ArgumentCaptor<TopicEntity> topicEntityCaptor = ArgumentCaptor.forClass(TopicEntity.class);
+
+        this.topicsService.save(this.topicsSupport.getModelFixtureMock(0));
+
+        verify(this.topicsMapper, times(1)).update(topicEntityCaptor.capture());
+        this.topicsSupport.assertEquals(this.topicsSupport.getModelFixtureMock(0), topicEntityCaptor.getValue());
+    }
+
+    @Test
+    public void testSaveWithOptions() throws Exception {
+        final TopicModelInterface topicModel = this.topicsSupport.getModelFixtureMock(0);
+        final TopicsOptionsInterface topicOptions = mock(TopicsOptionsInterface.class);
+
+        this.topicsService.save(topicModel, topicOptions);
+
+        verify(topicOptions, times(1)).saveWithRelations(topicModel);
+        this.assertServiceSet(topicOptions);
+    }
+
+    @Test
+    public void testDeleteIdentifiedModel() throws Exception {
+        final ArgumentCaptor<TopicEntity> topicEntityCaptor = ArgumentCaptor.forClass(TopicEntity.class);
+
+        this.topicsService.delete(this.topicsSupport.getModelFixtureMock(0));
+
+        verify(this.topicsMapper, times(1)).delete(topicEntityCaptor.capture());
+        this.topicsSupport.assertEquals(this.topicsSupport.getModelFixtureMock(0), topicEntityCaptor.getValue());
+    }
+
+    @Test
+    public void testDeleteUnidentifiedModelThrowsException() throws Exception {
+        final TopicModelInterface topicModel = this.topicsSupport.getModelAdditionalMock(0);
+
+        exception.expect(DeleteUnidentifiedModelException.class);
+        this.topicsService.delete(topicModel);
+    }
+
+    @Test
+    public void testDeleteWithOptions() throws Exception {
+        final TopicModelInterface topicModel = this.topicsSupport.getModelFixtureMock(0);
+        final TopicsOptionsInterface topicOptions = mock(TopicsOptionsInterface.class);
+
+        this.topicsService.delete(topicModel, topicOptions);
+
+        verify(topicOptions, times(1)).deleteWithRelations(topicModel);
+        this.assertServiceSet(topicOptions);
+    }
```

Comment:
Provide tests for:
 - save models list + withOptions
 - delete models list + with options
(look into subjects models test, there is one trick there)